src/messages/mod.rs

```diff
@@ -173,14 +173,20 @@ impl SignedRoutingMessage {
     /// Creates a `SignedMessage` with the given `content` and signed by the given `full_id`.
     pub fn new(
         content: RoutingMessage,
-        full_id: &FullId,
+        full_id: &BlsSecretKeyShare,
         pk_set: BlsPublicKeySet,
         proof: SectionProofChain,
     ) -> Result<SignedRoutingMessage> {
         let mut signatures = BTreeMap::new();
-        let pk_share = BlsPublicKeyShare(*full_id.public_id());
+        let pk_share = full_id.public_key_share();
+        // WIP: Get more efficient.
+        let position = (0..100)
```

Comment:
Wouldn't `(0..elder_size)` be enough? You would have to pass `elder_size` as argument though...

Reply:
Fixed with https://github.com/maidsafe/routing/pull/1913